sponsor-dapp-v2/src/views/ViewPositions.js

```diff
@@ -1,18 +1,152 @@
+import { drizzleReactHooks } from "drizzle-react";
+import TokenizedDerivative from "contracts/TokenizedDerivative.json";
+import { formatWei, formatWithMaxDecimals } from "common/FormattingUtils";
+import { useEtherscanUrl, useFaucetUrls } from "lib/custom-hooks";
 
 import Header from "components/common/Header";
 import Position from "components/Position";
 
-class ViewPositions extends Component {
-  render() {
-    const { landingPositions } = this.props;
+const BigNumber = require("bignumber.js");
 
-    if (!landingPositions) {
+function usePositionList() {
+  const { drizzle, useCacheCallPromise } = drizzleReactHooks.useDrizzle();
+  const { web3 } = drizzle;
+  const account = drizzleReactHooks.useDrizzleStatePromise((drizzleState, resolvePromise) => {
+    if (drizzleState.accounts[0]) {
+      resolvePromise(drizzleState.accounts[0]);
+    }
+  });
+
+  const registeredContracts = useCacheCallPromise("Registry", "getRegisteredDerivatives", account);
+
+  const finishedAddingContracts = drizzleReactHooks.useDrizzleStatePromise(
+    (drizzleState, resolvePromise, registeredContractsResolved) => {
+      let finished = true;
+
+      for (const registeredContract of registeredContractsResolved) {
+        const contractAddress = registeredContract.derivativeAddress;
+        if (!drizzleState.contracts[contractAddress]) {
+          finished = false;
+          drizzle.addContract({
+            contractName: contractAddress,
+            web3Contract: new drizzle.web3.eth.Contract(TokenizedDerivative.abi, contractAddress)
+          });
+        }
+      }
+
+      if (finished) {
+        resolvePromise(true);
+      }
+    },
+    [registeredContracts]
+  );
+
+  const etherscanUrl = useEtherscanUrl();
+
+  const positions = useCacheCallPromise(
+    "NotApplicable",
+    (
+      contractCall,
+      resolvePromise,
+      registeredContractsResolved,
+      accountResolved,
+      etherscanPrefix,
+      finishedAddingContracts
+    ) => {
+      let finished = true;
+
+      const call = (contractName, methodName, ...args) => {
+        const callResult = contractCall(contractName, methodName, ...args);
+        if (callResult === undefined) {
+          finished = false;
+        }
+        return callResult;
+      };
+
+      // Added a strange number as the fallback so it's obvious if this number ever makes it to the user.
+      const formatTokenAmounts = valInWei =>
+        valInWei ? formatWithMaxDecimals(formatWei(valInWei, web3), 4, false) : "-999999999";
+
+      const positions = registeredContractsResolved.map(registeredContract => {
+        const contractAddress = registeredContract.derivativeAddress;
+        const name = call(contractAddress, "name");
+        const totalSupply = formatTokenAmounts(call(contractAddress, "totalSupply"));
+        const yourSupply = formatTokenAmounts(call(contractAddress, "balanceOf", accountResolved));
+        const netPosition = BigNumber(yourSupply)
```

Comment:
Any reason why you prefer `BigNumber` here vs `web3.BN`?

Reply:
Not a huge preference. Since `web3.BN` can't do decimals, I would've had to do the subtraction on the raw output of `call`, handle the `undefined` output case, and then run format on the output of the subtraction. I figured it would be a little easier/simpler just to subtract the formatted outputs.

Reply:
Does that seem reasonable to you?